controllers/gcpmachine_controller_test.go

```diff
@@ -48,4 +63,171 @@ var _ = Describe("GCPMachineReconciler", func() {
 			Expect(result.RequeueAfter).To(BeZero())
 		})
 	})
+
+	Context("reconcile non-deleted GCPMachines", func() {
+		var (
+			reconciler   *GCPMachineReconciler
+			ctx          context.Context
+			clusterScope *scope.ClusterScope
+			machineScope *scope.MachineScope
+			mockCtrl     *gomock.Controller
+			gcpService   *mocks.MockServiceInterface
+		)
+		BeforeEach(func() {
+			flag.Set("logtostderr", "false")
+			flag.Set("v", "2")
+			klog.SetOutput(GinkgoWriter)
+			var err error
+			bootstrapData := "data"
+			version := "v1.16.2"
+
+			mockCtrl = gomock.NewController(GinkgoT())
+			gcpService = mocks.NewMockServiceInterface(mockCtrl)
+
+			reconciler = &GCPMachineReconciler{
+				Client:     k8sClient,
+				Log:        log.Log,
+				computeSvc: gcpService,
+			}
+
+			gcpMachine := &infrav1.GCPMachine{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "foo",
+					Namespace: "default",
+				},
+				Spec: infrav1.GCPMachineSpec{
+					Zone:         "us-west4-a",
+					InstanceType: "n1-standard-1",
+				},
+			}
+
+			machine := &clusterv1.Machine{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "fooMachine",
+					Namespace: "default",
+				},
+				Spec: clusterv1.MachineSpec{
+					InfrastructureRef: corev1.ObjectReference{
+						APIVersion: "infrastructure.cluster.x-k8s.io/v1alpha3",
+						Kind:       "GCPMachine",
+						Name:       gcpMachine.Name,
+					},
+					Bootstrap: clusterv1.Bootstrap{
+						Data: &bootstrapData,
+					},
+					Version: &version,
+				},
+			}
+
+			Expect(k8sClient.Create(ctx, gcpMachine)).Should(Succeed())
```

Comment:
Is there a need to use `k8sClient`? Can we just set up a `fakeClient = fake.NewFakeClient()` and use that everywhere?

Reply:
k8sclient is setup in the `suite_test` and is the recommended way in kubebuilder docs. I was already there as part of the existing test